javascriptlib/src/mill/javascriptlib/TestModule.scala

```diff
@@ -0,0 +1,147 @@
+package mill.javascriptlib
+
+import mill.*
+import os.*
+
+trait TestModule extends TaskModule {
+  import TestModule.TestResult
+
+  def test(args: String*): Command[Seq[TestResult]] =
+    Task.Command {
+      testTask(Task.Anon { args })()
+    }
+
+  def testCachedArgs: T[Seq[String]] = Task { Seq[String]() }
+
+  def testCached: T[Seq[TestResult]] = Task {
+    testTask(testCachedArgs)()
+  }
+
+  protected def testTask(args: Task[Seq[String]]): Task[Seq[TestResult]]
+
+  override def defaultCommandName() = "test"
+}
+
+object TestModule {
+  type TestResult = Unit
+
+  trait Shared extends TypeScriptModule {
+    override def upstreamPathsBuilder: T[Seq[(String, String)]] =
+      Task {
+        val stuUpstreams = for {
+          ((_, ts), mod) <- Task.traverse(moduleDeps)(_.compile)().zip(moduleDeps)
+        } yield (
+          mod.millSourcePath.subRelativeTo(Task.workspace).toString + "/test/utils/*",
+          (ts.path / "test" / "src" / "utils").toString
+        )
+
+        stuUpstreams ++ super.upstreamPathsBuilder()
+      }
+
+    def getPathToTest: T[String] = Task { compile()._2.path.toString }
+  }
+
+  trait Jest extends TypeScriptModule with Shared with TestModule {
+    override def npmDevDeps: T[Seq[String]] = Task {
+      Seq(
+        "@types/jest@^29.5.14",
+        "@babel/core@^7.26.0",
+        "@babel/preset-env@^7.26.0",
+        "jest@^29.7.0",
+        "ts-jest@^29.2.5",
+        "babel-jest@^29.7.0"
+      )
+    }
+
+    def testConfigSource: T[PathRef] =
+      Task.Source(millSourcePath / os.up / os.up / "jest.config.ts")
```

Comment:
Is this meant to be the repository root? If so it should be `Task.workspace / "jest.config.ts"`